### PR TITLE
Added dev scripts for utility purposes

### PR DESCRIPTION
**Pull request by yougotdirked**
- updated packages
- added packages as dev dependencies
```
nodemon
http-server
concurrently
```
- added npm script ``` npm start``` that uses above packages to register live changes.
- updated CONTRIBUTING.md

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,6 +2,14 @@
...
   "devDependencies": {
-    "@netlify/plugin-lighthouse": "^4.0.7"
+    "@netlify/plugin-lighthouse": "^4.0.7",
+    "http-server": "^14.1.1",
+    "nodemon": "^2.0.22"
+  },
+  "scripts": {
+    "build": "node build.js",
+    "build:watch": "nodemon -L -w template.html -w content build.js",
+    "start": "concurrently 'npm run build:watch' 'npm run serve'",
```
Comment (LucaScorpion) [inline package.json:12]: You probably need to add [concurrently](https://www.npmjs.com/package/concurrently) as a dev dependency, right?

**Review by LucaScorpion — CHANGES_REQUESTED** [review]
Looks good to me! Just one small thing that I think is missing from the `package.json`.

@jeroenheijmans Also a ping for you, since this is a not-insignificant change to the setup. With this the essentially "recommended" setup would mean using npm and installing some (dev) dependencies. I think overall it still simplifies things (and the local dev setup as is doesn't actually work well anymore), so to me it's worth it. But would also like your opinion on that :)

File: package.json (hunk 1 above)
Comment (yougotdirked) [reply package.json:13]: Done now :)

**Review by LucaScorpion — APPROVED** [review]

**Comment by jeroenheijmans** [comment]
In general I'm quite sad we're straying further and further from the "super simple html only" setup we used to have. Personally I would prefer to find a way back to less or even zero dependencies. For perspective:

- 0 dev dependencies => my preference
- 195 npm packages as dependencies => `main` at the time of writing
- 242 npm packages as dependencies => after this PR

Which also indicates that this PR doesn't change _much_, relatively. Besides, I feel like a majority of people would prefer convenience scripts and packages, and @LucaScorpion you also give your go ahead. So I'm happy to go along with that 😊 - perhaps some time soon (if I'm not too late by that time) I'll craft a proposal for removing all dependencies once again going back to bare-bones. But until that time, let's go ahead with this approach?

----

Having said that, I checked out this branch locally, ran `npm ci` and tried the documented `npm start`, but on Windows 10 it fails with this (version checks included for completeness):

```none
[yougotdirked-dev-utility-scripts] D:\git\infi\the-infi-way>  npm --version
8.18.0
[yougotdirked-dev-utility-scripts] D:\git\infi\the-infi-way>  node --version
v18.16.0
[yougotdirked-dev-utility-scripts] D:\git\infi\the-infi-way>  npm start

> start
> concurrently 'npm run build:watch' 'npm run serve'

[0] ''npm' is not recognized as an internal or external command,
[0] operable program or batch file.
[1] 'run' is not recognized as an internal or external command,
[1] operable program or batch file.
[2] The filename, directory name, or volume label syntax is incorrect.
[3] ''npm' is not recognized as an internal or external command,
[3] operable program or batch file.
[2] build:watch' exited with code 1
[1] run exited with code 1
[0] 'npm exited with code 1
[4] 'run' is not recognized as an internal or external command,
[4] operable program or batch file.
[5] 'serve'' is not recognized as an internal or external command,
[5] operable program or batch file.
[5] serve' exited with code 1
[4] run exited with code 1
[3] 'npm exited with code 1
```

With `npm run start` I get exactly the same issue. Ironically, these kinds of gotcha's is exactly what I worry about when introducing new dependencies 😅 (sorry, couldn't resist 😁).  I tried this with both Powershell and cmd.

I did some initial investigation on why this is happening and whether it's my own machine giving troubles. I'll try to find a moment to double check this on my work laptop later on, and am also open to more input. I believe @yougotdirked you also use a Windows laptop, right? Or are you on macOS/Linux with bash?

**Comment by jeroenheijmans** [comment]
We fixed some final things together, @yougotdirked will check with a colleague if it now also works on *nix and then merge this PR. 🎉

**Comment by LucaScorpion** [comment]
Works nicely on Linux still!

To address some of your concerns @jeroenheijmans:

> In general I'm quite sad we're straying further and further from the "super simple html only" setup we used to have. Personally I would prefer to find a way back to less or even zero dependencies.

I do fully agree with this. But currently (i.e. on `main`) the project doesn't actually run properly :grimacing: That is to say, the CSS doesn't load due to the way the path to that stylesheet is set up. Additionally, the steps in the contributing file didn't seem to work properly on Windows (or at least in Dirk's setup). While both of these are definitely things that we can fix without npm fluff, I think this having this is a fine temporary™ solution.

> I feel like a majority of people would prefer convenience scripts and packages

Hmmm, I'm not sure. I feel like _not_ having to install anything is also a big plus, but hard to say what the general consensus would be.

> perhaps some time soon (if I'm not too late by that time) I'll craft a proposal for removing all dependencies once again going back to bare-bones

I was also considering this :grin: So I'd fully support this.

**Pull request merged**